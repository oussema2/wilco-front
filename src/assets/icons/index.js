/* eslint-disable global-require */
export const aircraft = require( './aircraft.png' );
export const airport = require( './airport.png' );
export const arrowLeft = require( './arrowLeft.png' );
export const arrowRight = require( './arrowRight.png' );
export const award = require( './award.png' );
export const ban = require( './ban.png' );
export const bell = require( './bell.png' );
export const binoculars = require( './binoculars.png' );
export const brandSymbol = require( './brandSymbol.png' );
export const chevronLeft = require( './chevronLeft.png' );
export const chevronRight = require( './chevronRight.png' );
export const cloud = require( './cloud.png' );
export const certificate = require( './certificate.png' );
export const chatComment = require( './chatComment.png' );
export const chatNewMessage = require( './chatNewMessage.png' );
export const checkbox = require( './checkbox.png' );
export const checkboxUnselected = require( './checkboxUnselected.png' );
export const comment = require( './comment.png' );
export const cross = require( './cross.png' );
export const eye = require( './eye.png' );
export const eyeSlash = require( './eyeSlash.png' );
export const exclamationCircle = require( './exclamationCircle.png' );
export const exclamationTriangle = require( './exclamationTriangle.png' );
export const filter = require( './filter.png' );
export const flightInfoAircraft = require( './flightInfoAircraft.png' );
export const home = require( './home.png' );
export const horizontalLogo = require( './horizontalLogo.png' );
export const infoCircle = require( './infoCircle.png' );
export const logOut = require( './logOut.png' );
export const map = require( './map.png' );
export const meatballsMenu = require( './meatballsMenu.png' );
export const megaphone = require( './megaphone.png' );
export const onBoardingFirst = require( './onBoardingFirst.png' );
export const onBoardingSecond = require( './onBoardingSecond.png' );
export const onBoardingThird = require( './onBoardingThird.png' );
export const planeDeparture = require( './planeDeparture.png' );
export const plus = require( './plus.png' );
export const photoVideo = require( './photoVideo.png' );
export const privacyOnlyMe = require( './privacyOnlyMe.png' );
export const privacyPublic = require( './privacyPublic.png' );
export const remove = require( './remove.png' );
export const route = require( './route.png' );
export const rulerVertical = require( './rulerVertical.png' );
export const star = require( './star.png' );
export const starMyFeed = require( './starMyFeed.png' );
export const search = require( './search.png' );
export const settings = require( './settings.png' );
export const sliders = require( './sliders.png' );
export const speed = require( './speed.png' );
export const stopwatch = require( './stopwatch.png' );
export const thumbsUp = require( './thumbsUp.png' );
export const user = require( './user.png' );
export const users = require( './users.png' );
export const vector = require( './vector.png' );
