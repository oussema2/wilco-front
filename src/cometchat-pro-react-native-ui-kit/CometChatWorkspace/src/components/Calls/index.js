export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
