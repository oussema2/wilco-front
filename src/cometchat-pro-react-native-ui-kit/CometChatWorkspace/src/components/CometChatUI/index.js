/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
