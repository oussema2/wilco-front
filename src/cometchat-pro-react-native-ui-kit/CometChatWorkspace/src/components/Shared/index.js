export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
