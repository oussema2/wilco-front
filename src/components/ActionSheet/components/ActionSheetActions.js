import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
