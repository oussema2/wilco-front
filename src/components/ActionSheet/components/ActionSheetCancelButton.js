import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
