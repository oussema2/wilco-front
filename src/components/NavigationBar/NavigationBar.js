import React from 'react';
import {
	View, Text, TouchableOpacity
} from 'react-native';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react-lite';
import BackArrow from './BackArrow';
import { Image } from '../Image';
import { SuplementalTitle } from '../SuplementalTitle';
import { styles } from './NavigationBar.styles';
import { palette } from '../../Theme';

const NavigationBar = ( {
	testID, title, onBackArrowPress, onTitlePress, titleComponent, rightAction
} ) => {
	const _renderRightImage = () => (
		<TouchableOpacity onPress={rightAction.onPress}>
			<Image
				testID="rightIcon-testID"
				source={rightAction.image}
				style={{ ...styles.rightImageStyle, ...( rightAction.imageStyle || {} ) }}
				tintColor={palette.grayscale.black}
			/>
		</TouchableOpacity>
	);

	const _renderRightText = () => (
		<TouchableOpacity testID="onRightTextPress-testID" onPress={rightAction.onPress}>
			<SuplementalTitle testID="rightText-testID" text={rightAction.text} style={[ styles.rightTextStyle, rightAction.textStyle ]} />
		</TouchableOpacity>
	);

	return (
		<View testID={testID} style={styles.containerView}>
			<View style={[ styles.containerView, styles.backArrowContainer ]}>
				<BackArrow backArrowOnPress={onBackArrowPress} />
			</View>

			<View style={[ styles.containerView, styles.titleContainer ]}>

				{titleComponent || (
					<Text testID="title-text" style={styles.title} onPress={onTitlePress}>
						{title}
					</Text>
				)}

			</View>

			<View style={[ styles.containerView, styles.rightTextContainer ]}>
				<View testID="rightPlaceholder-view" style={( !rightAction.text ) ? styles.rightIconPlaceholder : null}>
					{ rightAction.onPress && rightAction.image && _renderRightImage() }
					{ rightAction.onPress && rightAction.text && !rightAction.image && _renderRightText() }
				</View>
			</View>
		</View>
	);
};
NavigationBar.propTypes = {
	testID: PropTypes.string,
	title: PropTypes.string,
	onBackArrowPress: PropTypes.func,
	onTitlePress: PropTypes.func,
	titleComponent: PropTypes.any,
	rightAction: PropTypes.shape( {
		onPress: PropTypes.func,
		image: PropTypes.node,
		imageStyle: PropTypes.any,
		text: PropTypes.string,
		textStyle: PropTypes.instanceOf( Object )
	} )
};

NavigationBar.defaultProps = {
	testID: 'navigationBar-component',
	title: '',
	onBackArrowPress: null,
	onTitlePress: null,
	titleComponent: null,
	rightAction: {}
};

export default observer( NavigationBar );
