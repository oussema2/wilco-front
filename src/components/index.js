export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
