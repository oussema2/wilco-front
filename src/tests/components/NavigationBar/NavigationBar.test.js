import * as React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native';
import { NavigationBar } from '../../../components/NavigationBar';
import { settings } from '../../../assets/icons';

afterEach( cleanup );

describe( 'NavigationBar', () => {
	let component;
	const onBackArrowPress = jest.fn();
	const onTitlePress = jest.fn();
	const onRightImagePress = jest.fn();
	const onRightTextPress = jest.fn();
	const rightImage = settings;
	const rightText = 'right Text';
	const title = 'Title';
	const testID = 'testing-navigationBar-component';

	const setUp = ( props ) => {
		jest.clearAllMocks();
		component = render(
			<NavigationBar onBackArrowPress={onBackArrowPress} title={title} testID={testID} {...props} />
		);
	};

	describe( 'with the default props', () => {
		it( 'renders the NavigationBar component correctly', () => {
			setUp();

			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component.queryByTestId( 'backArrow-image' ) ).toBeDefined();
			expect( component.queryByTestId( 'title-text' ) ).toHaveTextContent( title );
			expect( component.queryByTestId( 'rightPlaceholder-view' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'without a onBackArrowPress', () => {
		it( 'does not render the back arrow icon', () => {
			setUp( { onBackArrowPress: null } );

			expect( component.queryByTestId( 'backArrow-image' ) ).toBeNull();
		} );
	} );

	describe( 'with a onRightImagePress and rightImage', () => {
		it( 'renders the right icon', () => {
			setUp( { rightAction: { onPress: onRightImagePress, image: rightImage } } );

			expect( component.queryByTestId( 'rightIcon-testID' ) ).not.toBeNull();
		} );
	} );

	describe( 'without a onRightImagePress', () => {
		it( 'does not render the right icon', () => {
			setUp( { rightAction: { image: rightImage } } );

			expect( component.queryByTestId( 'rightIcon-testID' ) ).toBeNull();
		} );
	} );

	describe( 'without a rightImage', () => {
		it( 'does not render the right icon', () => {
			setUp( { rightAction: { onPress: onRightImagePress } } );

			expect( component.queryByTestId( 'rightIcon-testID' ) ).toBeNull();
		} );
	} );

	describe( 'with a onTitlePress', () => {
		it( 'renders the title correctly', () => {
			setUp( { onTitlePress } );
			expect( component.queryByTestId( 'title-text' ) ).toHaveTextContent( title );
		} );

		it( 'calls to onTitlePress when press the title', () => {
			setUp( { onTitlePress } );

			fireEvent.press( component.queryByTestId( 'title-text' ) );
			expect( onTitlePress ).toHaveBeenCalled();
		} );
	} );

	describe( 'without a onTitlePress', () => {
		it( 'renders the title correctly', () => {
			setUp( { onTitlePress } );
			expect( component.queryByTestId( 'title-text' ) ).toHaveTextContent( title );
		} );
	} );

	describe( 'with a rightText', () => {
		it( 'renders the NavigationBar correctly', () => {
			setUp( { rightAction: { onPress: onRightTextPress, text: rightText } } );
			expect( component.queryByTestId( 'rightText-testID' ) ).toHaveTextContent( rightText );
		} );
	} );

	describe( 'with a onRightTextPress', () => {
		it( 'calls to onRightTextPress when press the rightText', () => {
			setUp( { rightAction: { onPress: onRightTextPress, text: rightText, image: null } } );

			fireEvent.press( component.queryByTestId( 'onRightTextPress-testID' ) );
			expect( onRightTextPress ).toHaveBeenCalled();
		} );
	} );
} );
