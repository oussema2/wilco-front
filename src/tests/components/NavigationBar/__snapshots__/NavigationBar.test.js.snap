// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar with the default props renders the NavigationBar component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 44,
    }
  }
  testID="testing-navigationBar-component"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 44,
        },
        Object {
          "flex": 0.25,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/chevronLeft.png",
          }
        }
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
        testID="backArrow-image"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 44,
        },
        Object {
          "flex": 0.5,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      onPress={null}
      style={
        Object {
          "alignItems": "center",
          "color": "#000000",
          "flex": 1,
          "fontFamily": "NotoSansJP-Black",
          "fontSize": 16,
          "fontWeight": undefined,
          "letterSpacing": -0.3,
          "lineHeight": 22,
          "textAlign": "center",
        }
      }
      testID="title-text"
    >
      Title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 44,
        },
        Object {
          "flex": 0.25,
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "width": 20,
        }
      }
      testID="rightPlaceholder-view"
    />
  </View>
</View>
`;
